tests/driver/screenshot/build.rs

```diff
@@ -0,0 +1,159 @@
+pub fn collect_test_cases() -> std::io::Result<Vec<test_driver_lib::TestCase>> {
+    let mut results = vec![];
+
+    let case_root_dir: std::path::PathBuf =
+        [env!("CARGO_MANIFEST_DIR"), "..", "..", "screenshot"].iter().collect();
+
+    println!("cargo:rerun-if-env-changed=SLINT_TEST_FILTER");
+    let filter = std::env::var("SLINT_TEST_FILTER").ok();
+
+    for entry in walkdir::WalkDir::new(case_root_dir.clone()).follow_links(true) {
+        let entry = entry?;
+        let absolute_path = entry.into_path();
+        if absolute_path.is_dir() {
+            println!("cargo:rerun-if-changed={}", absolute_path.display());
+            continue;
+        }
+        let relative_path =
+            std::path::PathBuf::from(absolute_path.strip_prefix(&case_root_dir).unwrap());
+        if let Some(filter) = &filter {
+            if !relative_path.to_str().unwrap().contains(filter) {
+                continue;
+            }
+        }
+        if let Some(ext) = absolute_path.extension() {
+            if ext == "60" || ext == "slint" {
+                results.push(test_driver_lib::TestCase { absolute_path, relative_path });
+            }
+        }
+    }
+    Ok(results)
+}
+
+fn main() -> std::io::Result<()> {
+    let mut generated_file = std::fs::File::create(
+        Path::new(&std::env::var_os("OUT_DIR").unwrap()).join("generated.rs"),
+    )?;
+
+    for testcase in collect_test_cases()? {
+        println!("cargo:rerun-if-changed={}", testcase.absolute_path.display());
+        let mut module_name = testcase.identifier();
+        if module_name.starts_with(|c: char| !c.is_ascii_alphabetic()) {
+            module_name.insert(0, '_');
+        }
+        writeln!(generated_file, "#[path=\"{0}.rs\"] mod r#{0};", module_name)?;
+        let source = std::fs::read_to_string(&testcase.absolute_path)?;
+
+        let mut output = std::fs::File::create(
+            Path::new(&std::env::var_os("OUT_DIR").unwrap()).join(format!("{}.rs", module_name)),
+        )?;
+
+        #[cfg(not(feature = "build-time"))]
+        if !generate_macro(&source, &mut output, testcase)? {
+            continue;
+        }
+        #[cfg(feature = "build-time")]
+        generate_source(&source, &mut output, testcase)?;
+
+        for (i, x) in test_driver_lib::extract_test_functions(&source)
+            .filter(|x| x.language_id == "rust")
+            .enumerate()
+        {
+            write!(
+                output,
+                r"
+#[test] fn t_{}() -> Result<(), Box<dyn std::error::Error>> {{
+    use i_slint_backend_testing as slint_testing;
+    {}
+    Ok(())
+}}",
+                i,
+                x.source.replace('\n', "\n    ")
+            )?;
+        }
+    }
+
+    //Make sure to use a consistent style
+    println!("cargo:rustc-env=SLINT_STYLE=fluent");
+
+    println!("cargo:rustc-env=SLINT_EXPERIMENTAL_SYNTAX=true");
+
+    Ok(())
+}
+
+#[cfg(not(feature = "build-time"))]
+fn generate_macro(
```

Comment:
I don't think the macro will work with the SW renderer because we don't pass the flag to include image and fonts. So I'd leave that our